klippy/extras/tmc.py

```diff
@@ -158,11 +158,15 @@ def cmd_DUMP_TMC(self, params):
 
 # Endstop wrapper that enables "sensorless homing"
 class TMCVirtualEndstop:
-    def __init__(self, mcu_tmc, mcu_endstop):
+    def __init__(self, mcu_tmc, mcu_endstop, tmc_type):
+        self.tmc_type = tmc_type
         self.mcu_tmc = mcu_tmc
         self.fields = mcu_tmc.get_fields()
         self.mcu_endstop = mcu_endstop
-        self.en_pwm = self.fields.get_field("en_pwm_mode")
+        if self.tmc_type == 2130 or self.tmc_type == 5160:
+            self.en_pwm = self.fields.get_field("en_pwm_mode")
+        elif self.tmc_type == 2209:
+            self.en_pwm = self.fields.get_field("en_spreadCycle")
```

Comment:
I think it would be preferable for this code to just auto-detect the type (the same way that TMCStealthchopHelper does).

Reply:
se one comment above. i did that to be consistent at each point where field differ between the drivers. as emntioned one comment above i can remove that. i have no emotional bindings to consistency.

Reply:
I changed that to if reg solution.